openstudiocore/src/utilities/data/TimeSeries.cpp

```diff
@@ -626,11 +765,40 @@ namespace openstudio{
 
     std::shared_ptr<TimeSeries_Impl> TimeSeries_Impl::operator*(double d) const {
 
-      return std::shared_ptr<TimeSeries_Impl>(new TimeSeries_Impl(m_firstReportDateTime, 
-        m_secondsFromFirstReport,
-        m_values*d, 
+      return std::shared_ptr<TimeSeries_Impl>(new TimeSeries_Impl(m_startDateTime,
+        m_secondsFromStart,

```

Comment:
Yes, that's what I meant, but lets leave that alone until we work on
cleaning out TimeSeries. I've added a test of multiplication and changed
how operator* is working (again). Now a multiplied interval series stays an
interval series.

On Sat, May 23, 2015 at 10:40 AM, Dan Macumber <notifications@github.com>
wrote:

> In openstudiocore/src/utilities/data/TimeSeries.cpp
> <https://github.com/NREL/OpenStudio/pull/1489#discussion_r30946183>:
>
> > @@ -626,11 +765,40 @@ namespace openstudio{
> >
> >      std::shared_ptr<TimeSeries_Impl> TimeSeries_Impl::operator*(double d) const {
> >
> > -      return std::shared_ptr<TimeSeries_Impl>(new TimeSeries_Impl(m_firstReportDateTime,
> > -        m_secondsFromFirstReport,
> > -        m_values*d,
> > +      return std::shared_ptr<TimeSeries_Impl>(new TimeSeries_Impl(m_startDateTime,
> > +        m_secondsFromStart,
>
> Sure, we can change impl interfaces, is that what you are asking? I'm
> guessing we need a combination of more tests (e.g. make sure multiplying by
> constant doesnt change dates) and simplified impl state (so there is less
> chance for confusion). Think we can do that next week?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NREL/OpenStudio/pull/1489/files#r30946183>.
>